python/sparkdl/graph/graph_factory.py

```diff
@@ -0,0 +1,136 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import logging
+from pathlib import Path
+import shutil
+import six
+from tempfile import mkdtemp
+
+import tensorflow as tf
+
+from .graph_builder import IsolatedSession
+from ..utils import graph_utils as tfx
+from ..image.imageIO import SparkMode
+
+logger = logging.getLogger('sparkdl')
+
+"""
+Build various pieces of the function
+
+TODO: We might want to cache some of the big models in their GraphFunction format
+      Deserializing ProtocolBuffer bytes is in general faster than directly loading Keras models.
+"""
+
+def build_spimage_converter(img_dtype):
+    """
+    Convert a imageIO byte encoded image into a image tensor suitable as input to ConvNets
+    The name of the input must be a subset of those specified in `image.imageIO.imageSchema`.
+
+    :param img_dtype: the type of data the underlying image bytes represent
+    """
+    with IsolatedSession() as issn:
+        # Flat image data -> image dimensions
+        # This has to conform to `imageIO.imageSchema`
+        height = tf.placeholder(tf.int32, [], name="height")
+        width = tf.placeholder(tf.int32, [], name="width")
+        num_channels = tf.placeholder(tf.int32, [], name="nChannels")
+        image_buffer = tf.placeholder(tf.string, [], name="data")
+
+        # The image is packed into bytes with height as leading dimension
+        # This is the default behavior of Python Image Library
+        shape = tf.reshape(tf.stack([height, width, num_channels], axis=0),
+                           shape=(3,), name='shape')
+        if SparkMode.RGB == img_dtype:
+            image_uint8 = tf.decode_raw(image_buffer, tf.uint8, name="decode_raw")
+            image_float = tf.to_float(image_uint8)
+        else:
+            assert img_dtype == SparkMode.RGB_FLOAT32, \
+                "Unsupported dtype for image: {}".format(img_dtype)
+            image_float = tf.decode_raw(image_buffer, tf.float32, name="decode_raw")
+
+        image_reshaped = tf.reshape(image_float, shape, name="reshaped")
+        image_input = tf.expand_dims(image_reshaped, 0, name="image_input")
+        gfn = issn.asGraphFunction([height, width, image_buffer, num_channels], [image_input])
+
+    return gfn
+
+def build_identity():
+    with IsolatedSession() as issn:
+        pred_input = tf.placeholder(tf.float32, [None, None])
+        final_output = tf.identity(pred_input, name='output')
+        gfn = issn.asGraphFunction([pred_input], [final_output])
+
+    return gfn
+
+def build_flattener():
+    with IsolatedSession() as issn:
+        mat_input = tf.placeholder(tf.float32, [None, None])
+        mat_output = tf.identity(tf.reshape(mat_input, shape=[-1]), name='output')
+        gfn = issn.asGraphFunction([mat_input], [mat_output])
+
+    return gfn
+
+def merge(functions):
```

Comment:
is it too messy to put in GraphFunction as e.g. `from_graph_functions` ? it's a little out of place in this module.